cmd/single-ocsp/test/gentest.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/json"
+	"encoding/pem"
+	"io/ioutil"
+	"math/big"
+	"time"
+
+	"github.com/cloudflare/cfssl/crypto/pkcs11key"
+)
+
+const pkcs11FileName = "pkcs11.json"
+const caCertFileName = "ca-cert.der"
+const eeCertFileName = "ee-cert.der"
+const ocspCertFileName = "ocsp-cert.der"
+const caCertFileNamePEM = "ca-cert.pem"
+const eeCertFileNamePEM = "ee-cert.pem"
+const ocspCertFileNamePEM = "ocsp-cert.pem"
+
+func panicOnError(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+func makeCert(template, issuer *x509.Certificate, pub interface{}, priv crypto.Signer) *x509.Certificate {
+	// Always use pretty much the same notBefore and notAfter
```

Comment:
This comment contradicts the below text. I think probably hardcoding specific dates (with a far-far-future NotAfter) is best to make this test non-flaky. In particular, the `.Round(time.Hour)` below is an anti-pattern. I recently fixed a bug in cfssl that was using `Round` instead of `Truncate`, causing test flakiness depending on which half of the hour you ran it in.